src/components/roomStack/FlippedStackRoom.tsx

```diff
@@ -17,8 +22,23 @@ const FlippedStackRoom: FunctionComponent<FlippedStackRoomProps> = ({
   doorDirections,
 }) => {
   const roomBox = getRoomBoundingBox(areaBox);
+  const dispatch = useDispatch();
+  const gridSize = useGridSize();
+  const boardTopLeft = useSelector((state: RootState) => state.board.topLeft);
+  const [groupTopLeft, setGroupTopLeft] = useState<Point>({ x: 0, y: 0 });
+
   return (
-    <Group>
+    <Group
+      draggable
+      onDragEnd={(e) => {
+        const droppedOn = droppedOnGrid(e, roomBox, gridSize, boardTopLeft);
+        dispatch(flippedRoomDropped(droppedOn));
+        setGroupTopLeft({ x: e.target.x(), y: e.target.y() });
+        setGroupTopLeft({ x: 0, y: 0 });
```

Comment:
This is a janky way to get the room to snap back to the stack if it isn't dropped on a valid spot. If the component is re-rendered and the `x` and `y` of a Konva element hasn't changed, Konva remembers where it was. I assume this is to avoid re-renders making dragging impossible. Setting its position to where it was dropped before setting it back to (0, 0) fixes this.

As an aside, I set it back to (0, 0) because it seems that coordinates inside `Group`s are relative; that is, children of `Group`s are drawn at coordinates relative to that of the parent `Group`. Using this could eliminate some need for knowing the global position of every element everywhere. I'll keep that in mind.